Integrate external sru docs

dviererbe opened the pull request:
This PR integrates and refers to the new SRU documentation: https://canonical-sru-docs.readthedocs-hosted.com/en/latest/

The dedicated SRU doc (introduced with #69) is moved to `explanation/development-process` and reduced to the most important information, just to give a brief overview of the process.

dviererbe committed:
chore: enforce max line length of 80 characters

dviererbe committed:
fix: remove multiple occurence of term 'Ubuntu SRU Team'

dviererbe committed:
feat: add intersphinx ref to official SRU docs

dviererbe committed:
feat: incorporate external SRU documentation

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
-
-The dedicated :doc:`/explanation/stable-release-updates` article describes
-these criteria and the procedure for preparing such an update.
+After publication of a :term:`Ubuntu Stable Release`, there may be a need
```
```suggestion
After publication of an :term:`Ubuntu Stable Release`, there may be a need
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
-The dedicated :doc:`/explanation/stable-release-updates` article describes
-these criteria and the procedure for preparing such an update.
+After publication of a :term:`Ubuntu Stable Release`, there may be a need
+to update it or fix bugs. You can fix these newly discovered bugs and make
```
```suggestion
to update it or fix bugs. You can fix these newly-discovered bugs and make
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+updates through a special process known as **Stable Release Update (SRU)**.
+
+The SRU process ensures that any changes made to a stable release are thoroughly
+vetted and tested before being made available to users. This is because many of
+the users rely on the stability of the stable release for their day-to-day
```
```suggestion
vetted and tested before being made available to users. This is because many
users rely on the stability of the stable release for their day-to-day
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+operations, and any problem they experience with it can be disruptive.
+
+The following paragraphs intend to give you a brief introduction to the SRU
+process. See the decicated :external+sru:ref:`Ubuntu SRU Documentation <home>`
```
```suggestion
process. See the dedicated :external+sru:ref:`Ubuntu SRU Documentation <home>`
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+1. Ensure the bug is fixed in the :term:`current development release
+   <Current Release in Development>` and all subsequent supported releases to
+   ensure consistency across different Ubuntu versions, especially preventing
+   regressions when users upgrade to newer ones.
```
```suggestion
   regressions when users upgrade to newer releases.
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+#. Get the package with the SRU patch into the upload queue.
+#. The SRU team will then review from the unapproved queue. When the upload is
+   ready, the SRU team accepts the upload into the proposed pocket.
+#. Once the builds are ready, autopkgtest are triggered. Test the binaries
```
Do we have a good link that defines more about what autopkgtests are?

rkratky replied:
The resource will be the article in this guide (once it's added ;)): https://canonical-ubuntu-packaging-guide.readthedocs-hosted.com/en/latest/how-to/run-tests/. In the meantime, the info is all over the place:

- https://wiki.ubuntu.com/ProposedMigration#autopkgtests
- https://ubuntu-packaging-guide.readthedocs.io/en/latest/ubuntu-packaging-guide/auto-pkg-test.html
- https://wiki.debian.org/ContinuousIntegration/autopkgtest
- https://people.debian.org/~eriberto/README.package-tests.html
- ...

dviererbe replied:
A few lines above there is a glossary entry reference to the term `autopkgtest`. I added relevant terms to the glossary for this topic.

See: b556d58383859e79c24eb5284a518f39a8864241

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+verified by the reporter or affected users of the SRU bug in a software 
+environment that closely resembles the state after the SRU team copies the
+package to the updates pocket. Generally, this will be with a system that's up
+to date from the release, security, and updates pockets. It shouldn't include
```
```suggestion
to date with the release, security, and updates pockets. It shouldn't include
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+environment that closely resembles the state after the SRU team copies the
+package to the updates pocket. Generally, this will be with a system that's up
+to date from the release, security, and updates pockets. It shouldn't include
+other packages from the proposed or backports pocket, except commonly installed
```
```suggestion
other packages from the proposed or backports pocket, except commonly-installed
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+other packages from the proposed or backports pocket, except commonly installed
+packages built from the affected source package.
+
+Read more about this process :external+sru:doc:`here <howto/release>`.
```
```suggestion
Read :external+sru:doc:`more about this process <howto/release>`.
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+^^^^^^^^^^^
+
+Once a package is released to the updates pocket, the update is then phased 
+so that the update is gradually made available to expanding subsets of Ubuntu
```
```suggestion
so it is gradually made available to expanding subsets of Ubuntu
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+so that the update is gradually made available to expanding subsets of Ubuntu
+users.
+
+Read more about this process :external+sru:ref:`here <explanation-phasing>`.
```
```suggestion
Read :external+sru:ref:`more about this process <explanation-phasing>`.
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+^^^^^^^^^^^
+
+Regressions are unintended negative consequences that updates introduce.
+They appear as new bugs or failures in previously well-functioning aspect of an 
```
```suggestion
They appear as new bugs or failures in previously well-functioning aspects of an 
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+They appear as new bugs or failures in previously well-functioning aspect of an 
+Ubuntu release. 
+
+Read more about regressions :external+sru:ref:`here <explanation-regressions>`
```
```suggestion
Read :external+sru:ref:`more about regressions <explanation-regressions>`
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+Ubuntu release. 
+
+Read more about regressions :external+sru:ref:`here <explanation-regressions>`
+and how to handle them :external+sru:ref:`here <howto-report-regression>`.
```
```suggestion
and :external+sru:ref:`how to handle them <howto-report-regression>`.
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+^^^^^^^^^^^^^^^
+
+If a bug fixed by an update doesn't get any testing or verification feedback for
+90 days, an automated call for testing comment will be made on the bug report.
```
```suggestion
90 days, an automated "call for testing" comment will be made on the bug report.
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+testing, the :term:`Stable Release Managers` will remove the package from
+proposed and close the bug task as ``Won't Fix``.
+
+Also, updates will be removed from proposed if they introduce a nontrivial
```
```suggestion
Also, updates will be removed from proposed if they introduce a non-trivial
```

s-makin commented on docs/explanation/stable-release-updates.rst:
```diff
@@ -1,249 +0,0 @@
-Stable Release Updates (SRU)
```
This page being removed has introduced a broken reference in the "debian-merges-and-syncs" page at line 42. You'll need to remove that reference on that page as well, it's currently preventing RTD from building the preview.

dviererbe replied:
Fixed with: 4457e5cbaba78569ef7396acde5426d8c6472eab

s-makin commented on docs/reference/glossary.rst:
```diff
@@ -1327,8 +1327,8 @@ Glossary
     Ubuntu Stable Release
-        The Ubuntu stable release is the officially published version of Ubuntu and
-        its set of :term:`packages <Package>`.
+        Ubuntu stable releases are officially published versions of Ubuntu
```
```suggestion
        Ubuntu stable releases are officially-published versions of Ubuntu
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+Overview
+~~~~~~~~
+
+A typical SRU will be perfomed like this:
```
```suggestion
A typical SRU will be performed like this:
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+   in the :term:`Ubuntu Archive` and follow up in the bug report with your 
+   verification results.
+#. The Ubuntu SRU Team will evaluate the testing feedback and move the package
+   into :ref:`updates <ArchivePockets_Updates>` after it passes a minimum aging
```
```suggestion
   into :ref:`updates <ArchivePockets_Updates>` after it passes a minimum ageing
```

s-makin commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+
+If a bug fixed by an update doesn't get any testing or verification feedback for
+90 days, an automated call for testing comment will be made on the bug report.
+If no testing occurs within an additional 15 days, totaling 105 days without any
```
```suggestion
If no testing occurs within an additional 15 days, totalling 105 days without any
```

s-makin reviewed (commented):
Thanks, generally looks good! I've spotted some nits, and some other things that are blocking the automated checks. In addition to the inline comments:
1) You'll want to add "SRUs" and "StableReleaseUpdates" to the spelling ignore list
2) Deleting the `stable-release-udpates.rst` file has broken a link in `debian-merges-and-syncs.rst` at line 42
3) There are two links that say "broken" - they're not really broken, but they are redirecting so we probably want to update them:

- https://ubuntu.com/community/code-of-conduct --> https://ubuntu.com/community/ethos/code-of-conduct
- https://ubuntu.com/server/docs/package-management#advanced-packaging-tool --> https://documentation.ubuntu.com/server/how-to/software/package-management/

dviererbe committed:
fix: spelling, grammar, wording

Co-authored-by: Sally <sally.makin@canonical.com>

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+Overview
+~~~~~~~~
+
+A typical SRU will be performed like this:
```
```suggestion
A typical SRU is performed like this:
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/stable-release-updates.rst:
The removal of this file causes a breaking error:

```
docs/explanation/debian-merges-and-syncs.rst:42:
      WARNING: unknown document: '/explanation/stable-release-updates' [ref.doc]
```

dviererbe replied:
Fixed with: 4457e5cbaba78569ef7396acde5426d8c6472eab

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+Once the SRU team accepts the SRU into the proposed pocket, the SRU has to be
+verified by the reporter or affected users of the SRU bug in a software 
+environment that closely resembles the state after the SRU team copies the
+package to the updates pocket. Generally, this will be with a system that's up
```
```suggestion
package to the updates pocket. Generally, this is with a system that's up
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+SRU phasing
+^^^^^^^^^^^
+
+Once a package is released to the updates pocket, the update is then phased 
```
```suggestion
Once a package is released to the updates pocket, the update is then phased, 
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+so it is gradually made available to expanding subsets of Ubuntu
+users.
+
+Read :external+sru:ref:`more about this process <explanation-phasing>`.
```
```suggestion
Read :external+sru:ref:`more about phasing <explanation-phasing>`.
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+Ubuntu release. 
+
+Read :external+sru:ref:`more about regressions <explanation-regressions>`
+and :external+sru:ref:`how to handle them <howto-report-regression>`.
```
```suggestion
and :external+sru:ref:`how to handle regressions <howto-report-regression>`.
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+^^^^^^^^^^^^^^^
+
+If a bug fixed by an update doesn't get any testing or verification feedback for
+90 days, an automated "call for testing" comment will be made on the bug report.
```
```suggestion
90 days, an automated "call for testing" comment is made on the bug report.
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+If a bug fixed by an update doesn't get any testing or verification feedback for
+90 days, an automated "call for testing" comment will be made on the bug report.
+If no testing occurs within an additional 15 days, totalling 105 days without any
+testing, the :term:`Stable Release Managers` will remove the package from
```
```suggestion
testing, the :term:`Stable Release Managers` removes the package from
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+testing, the :term:`Stable Release Managers` will remove the package from
+proposed and close the bug task as ``Won't Fix``.
+
+Also, updates will be removed from proposed if they introduce a non-trivial
```
```suggestion
Also, updates are removed from proposed if they introduce a non-trivial
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+#. Once the builds are ready, autopkgtest are triggered. Test the binaries
+   in the :term:`Ubuntu Archive` and follow up in the bug report with your 
+   verification results.
+#. The Ubuntu SRU Team will evaluate the testing feedback and move the package
```
```suggestion
#. The Ubuntu SRU Team evaluates the testing feedback and moves the package
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky commented on docs/explanation/development-process.rst:
```diff
@@ -312,9 +312,111 @@ the "Final Release", a representative of the team announces it on the
+   Plan to verify that the bug was fixed and highlight where problems could
+   occur.
+#. Get the package with the SRU patch into the upload queue.
+#. The SRU team will then review from the unapproved queue. When the upload is
```
```suggestion
#. The SRU team then reviews from the unapproved queue. When the upload is
```

dviererbe replied:
Resolved with: 2d97be5f0cbe0ba604812c9ef07ae53d81f6a7a5

rkratky reviewed (changes requested):
Thanks, @dviererbe! Some nitpicks and a necessary fix (updating the link to the removed SRU file).

dviererbe committed:
fix: broken link

The removal of the file `docs/explanation/stable-release-updates.rst`
caused a broken link. This commit fixes the link.

dviererbe committed:
feat: incorporate wording improvements

Co-authored-by: Robert Krátký <rkratky@gmail.com>

dviererbe committed:
feat: autopkgtest related glossary entries

Add glossary entries for teh terms:
- autopkgtest
- autopkgtest Cloud
- Debian Enhancement Proposal
- DEP
- DEP 8

dviererbe committed:
fix(spelling): add SRU related words to wordlist

dviererbe commented:
@rkratky I incorporated your and Sallys feedback and fixed false negatives in the spellcheck. There are two links that say "broken", but they're not really broken. **Can you re-review this PR?**

rkratky reviewed (approved):
Thanks, @dviererbe.
